hack/generator/pkg/codegen/pipeline_determine_resource_ownership.go

```diff
@@ -0,0 +1,290 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package codegen
+
+import (
+	"context"
+	"github.com/Azure/k8s-infra/hack/generator/pkg/astmodel"
+	"github.com/pkg/errors"
+	"strings"
+)
+
+const resourcesPropertyName = astmodel.PropertyName("Resources")
+
+func determineResourceOwnership() PipelineStage {
+	return PipelineStage{
+		"Determine ARM resource relationships",
+		func(ctx context.Context, definitions astmodel.Types) (astmodel.Types, error) {
+			return determineOwnership(definitions)
+		},
+	}
+}
+
+func determineOwnership(definitions astmodel.Types) (astmodel.Types, error) {
+
+	updatedDefs := make(astmodel.Types)
+
+	for _, def := range definitions {
+		if resourceType, ok := def.Type().(*astmodel.ResourceType); ok {
```

Comment:
Maybe we should add a convenience method for this on the Type interface? We do it all over the place. I might do that after this lands.

Reply:
Yeah, might be worth doing but maybe split into a separate PR so we can see that it found all the usages where we're doing this?

Doing it in this PR I think we'd miss that for all the other changes here.